src/metatensor/models/utils/per_atom.py

```diff
@@ -1,5 +1,40 @@
+from typing import Dict, List
+
 import torch
 from metatensor.torch import TensorBlock, TensorMap
+from metatensor.torch.atomistic import System
+
+
+def average_by_num_atoms(
+    tensor_map_dict: Dict[str, TensorMap],
+    systems: List[System],
+    per_structure_keys: List[str],
+):
+    """
+    Averages a dictionary of :py:class:`TensorMap` objects by the number of
```

Comment:
are you sure that :py:class:\`TensorMap\` will be rendered with the correct link? Otherwise you might just use 
```suggestion
    Averages a dictionary of ``TensorMap`` objects by the number of
```